### PR TITLE
CB-10080 Rewrite file-transfer server not to use disk cache

### PR DESCRIPTION
**Pull request by daserge**
Used [busboy](https://github.com/mscdex/busboy) instead of formidable

[Jira issue](https://issues.apache.org/jira/browse/CB-10080)

### CODE REVIEW

**Comment by daserge** [comment]
@dblotsky, @jasongin, @stevengill, could you please take a look?

File: server.js [hunk 1]
```diff
@@ -11,6 +12,54 @@ var LATIN1_SYMBOLS = '¥§©ÆÖÑøøø¼';
...
 
+function parseMultipartForm(req, res, finishCb) {
+    var fields = {}, files = {};
+
+    var busboy = new Busboy({ headers: req.headers });
+    busboy.on('file', function(fieldname, file, filename, encoding, mimetype) {
+        var currentFile = {};
+        file.on('data', function(data) {
+            currentFile.name = filename;
+            currentFile.size = data.length;
```
Comment (dblotsky) [inline server.js]: This guy should be `+=` instead of `=`.

File: server.js [hunk 2]
```diff
@@ -11,6 +12,54 @@ var LATIN1_SYMBOLS = '¥§©ÆÖÑøøø¼';
 var Iconv  = require('iconv').Iconv;
 var iconv = new Iconv('UTF-8', 'ISO-8859-1');
 
+function parseMultipartForm(req, res, finishCb) {
+    var fields = {}, files = {};
+
+    var busboy = new Busboy({ headers: req.headers });
+    busboy.on('file', function(fieldname, file, filename, encoding, mimetype) {
+        var currentFile = {};
```
Comment (dblotsky) [inline server.js]: This guy should be `{size: 0}` to start.

File: server.js [hunk 3]
```diff
@@ -11,6 +12,54 @@ var LATIN1_SYMBOLS = '¥§©ÆÖÑøøø¼';
...
+
+    busboy.on('finish', function() {
+        console.log(stringify({fields: fields, files: files}));
+
+        finishCb(req, res, {fields: fields, files: files});
+    });
+
+    busboy.on('error', function(err) {
+        console.error('error: ' + err + ': ' + JSON.stringify(err));
+        req.connection.destroy();
```
Comment (dblotsky) [inline server.js]: Might it be better to let the caller handle the error, instead of closing the connection as a side effect?

**Comment by daserge** [comment]
@dblotsky, thanks for reviewing!
[Addressed](https://github.com/daserge/cordova-labs/commit/091d64c57cb35ce3fe21042821abb30aad2bcb2a), please take a look.

**Comment by dblotsky** [comment]
LGTM. Thanks for addressing the comments!

**Comment by daserge** [comment]
Merged. @dblotsky, @stevengill, could you please update the server?

**Comment by stevengill** [comment]
@dblotsky, did we get the server back up?

**Comment by stevengill** [comment]
@dblotsky Did you get the server back up after you got the VM rebuilt?

**Pull request merged**